openshift-knative-operator/pkg/serving/secretinformerfilterting_test.go

```diff
@@ -185,101 +123,15 @@ func TestSecretInformerFilteringOverride(t *testing.T) {
-			ks.Spec.DeploymentOverride = append(ks.Spec.DeploymentOverride, base.WorkloadOverride{
-				Name: "net-istio-controller",
-				Env: []base.EnvRequirementsOverride{
-					{
-						Container: "controller",
-						EnvVars: []corev1.EnvVar{{
-							Name:  EnableSecretInformerFilteringByCertUIDEnv,
-							Value: "false",
-						}, {
-							Name:  "foo",
-							Value: "foo",
-						}},
-					}},
-			})
-		}),
-		expected: ks(func(ks *operatorv1beta1.KnativeServing) {
-			ks.Spec.Ingress = &operatorv1beta1.IngressConfigs{Istio: base.IstioIngressConfiguration{
-				Enabled: true,
-			}}
-			ks.Annotations = map[string]string{secretInformerFilteringAnnotation: "true"}
-			ks.Spec.DeploymentOverride = append(ks.Spec.DeploymentOverride, base.WorkloadOverride{
-				Name: "net-istio-controller",
-				Env: []base.EnvRequirementsOverride{
-					{
-						Container: "controller",
-						EnvVars: []corev1.EnvVar{{
-							Name:  EnableSecretInformerFilteringByCertUIDEnv,
-							Value: "false",
-						}, {
-							Name:  "foo",
-							Value: "foo",
-						}},
-					}},
-			})
-		}),
-		shouldAddLabelToSecret: false,
-	}, {
-		name: "deprecated annotation is set to false but env var takes precedence, enabled secret filtering, istio enabled",
-		in: ks(func(ks *operatorv1beta1.KnativeServing) {
-			ks.Spec.Ingress = &operatorv1beta1.IngressConfigs{Istio: base.IstioIngressConfiguration{
-				Enabled: true,
-			}}
-			ks.Annotations = map[string]string{secretInformerFilteringAnnotation: "false"}
-			ks.Spec.DeploymentOverride = append(ks.Spec.DeploymentOverride, base.WorkloadOverride{
-				Name: "net-istio-controller",
-				Env: []base.EnvRequirementsOverride{
-					{
-						Container: "controller",
-						EnvVars: []corev1.EnvVar{{
-							Name:  EnableSecretInformerFilteringByCertUIDEnv,
-							Value: "true",
-						}, {
-							Name:  "foo",
-							Value: "foo",
-						}},
-					}},
-			})
-		}),
-		expected: ks(func(ks *operatorv1beta1.KnativeServing) {
-			ks.Spec.Ingress = &operatorv1beta1.IngressConfigs{Istio: base.IstioIngressConfiguration{
-				Enabled: true,
-			}}
-			ks.Annotations = map[string]string{secretInformerFilteringAnnotation: "false"}
-			ks.Spec.DeploymentOverride = append(ks.Spec.DeploymentOverride, base.WorkloadOverride{
-				Name: "net-istio-controller",
-				Env: []base.EnvRequirementsOverride{
-					{
-						Container: "controller",
-						EnvVars: []corev1.EnvVar{{
-							Name:  EnableSecretInformerFilteringByCertUIDEnv,
-							Value: "true",
-						}, {
-							Name:  "foo",
-							Value: "foo",
-						}},
-					}},
-			})
-		}),
-		shouldAddLabelToSecret: false,
 	}}
 
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
 			tf := enableSecretInformerFilteringTransformers(c.in)
-			if c.shouldAddLabelToSecret {
-				if tf == nil {
-					t.Errorf("Secret transformer should not be nil")
-				}
+			if c.shouldAddLabelToSecret && tf == nil {
```

Comment:
Hm, I'd say the old version was more readable. The else if part is a bit hard to parse.

Reply:
This part was not tested in the past. We missed that case so it is complete now.